spec/controllers/api/documents_controller_spec.rb

```diff
@@ -26,28 +30,65 @@
       let(:params) { {} }
 
       it 'returns all documents' do
-        expect(meta_data['results']).to eq 4
-        expect(documents.count).to eq 4
+        expect(meta_data['results']).to eq 7
+        expect(documents.count).to eq 7
       end
     end
 
     context 'when documents of type insight are requested' do
       let(:params) { { document_type: 'insight' } }
       it 'returns all insight documents' do
-        expect(meta_data['results']).to eq 3
-        expect(documents.count).to eq 3
+        expect(meta_data['results']).to eq 6
+        expect(documents.count).to eq 6
       end
     end
   end
 
   describe 'keyword search' do
     context 'when the document_type is specified' do
       context 'when a keyword is provided' do
-        let(:params) { { document_type: 'insight', keyword: 'pension' } }
+        context 'and the keyword is found in the content' do
+          let(:params) { { document_type: 'insight', keyword: 'pension' } }
 
-        it 'returns an array of documents which contain the keyword' do
-          expect(meta_data['results']).to eq 1
-          expect(documents.count).to eq 1
+          it 'returns an array of documents which contain the keyword' do
+            expect(meta_data['results']).to eq 1
+            expect(params[:keyword]).to be_in(
+              documents.first["blocks"].first["content"]
+            )
+          end
+        end
+
+        context 'and the keyword is found in the middle of the title' do
+          let(:params) { { document_type: 'insight', keyword: 'stress' } }
+
+          it 'returns documents with the keyword in the title' do
+            expect(params[:keyword]).to be_in(documents.first["label"])
+          end
+        end
+
+        context 'and the keyword is found at the start of the title' do
+          let(:params) { { document_type: 'insight', keyword: 'annuity' } }
+
+          it 'returns documents with the keyword at the start of the title' do
+            expect(documents.first["label"]).to start_with(params[:keyword])
+          end
+        end
+
+        context 'when the keyword is in the title of one document and content of another' do
+          let(:params) { { document_type: 'insight', keyword: 'annuity' } }
+
+          it 'returns both documents' do
+            documents.each do |doc|
+              doc_title = doc['label']
+              doc_content = doc['blocks'].select do |block|
+                block['identifier'] =='content' && block['content'].include?(params[:keyword])
```

Comment:
as an alternative to my suggestion above, if you change this line to just be
`block['identifier'] =='content'` as you don't need to check for the keyword in here, that is the responsibility of the search and thus the expectation below.